src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -386,7 +388,19 @@ public void write() {
             }
           case SHOULD_ROTATE:
             updateRotationTimers(currentRecord);
-            closeTempFile();
+            try {
+              closeTempFile();
+            } catch (Exception e) {
+              log.error("Failed to close temp file.",e);
+              if (this.failedTempFileCloseAttempts < 4) {
```

Comment:
`4` should be configurable, and if not configurable it should be a constant instead of a literal. 4 can be the default of the configuration.

Reply:
And if this condition is false, what then? You just captured the exception, and the code will continue as if nothing happened. As you wrote it, it should throw a new exception with `e` as the cause, and make sure the outer `try` will rethrow it instead of retrying.

Reply:
4 will be made a constant.
For the re throwing case, will try to move this handling to the outside try catch block and if the retries are exhausted, we will retain the default behavior of just setting the retry timeout and breaking from the loop.